s3_scrubber/src/checks.rs

```diff
@@ -136,24 +135,17 @@ pub(crate) async fn branch_cleanup_and_check_errors(
                             // a new generation that didn't upload an index yet.
                             //
                             // Even so, a layer that is not referenced by the index could just
-                            // be something enqueued for deletion, so while this check is valid 
+                            // be something enqueued for deletion, so while this check is valid
                             // for indicating that a layer is garbage, it is not an indicator
                             // of a problem.
                             gen < &index_part_generation)
                         .collect();
 
+                    // unlinked files at the current generation are not considered to be deleted,
+                    // because a pageserver could still have those in its deletion queue or even as
+                    // layers which are being read.
+
                     if !orphan_layers.is_empty() {
-                        result.errors.push(format!(
-                            "index_part.json does not contain layers from S3: {:?}",
-                            orphan_layers
-                                .iter()
-                                .map(|(layer_name, gen)| format!(
-                                    "{}{}",
-                                    layer_name.file_name(),
-                                    gen.get_suffix()
-                                ))
-                                .collect::<Vec<_>>(),
-                        ));
```

Comment:
Why did this need to go?

Reply:
Does not look familiar, need to check, I did do the merge main.

Reply:
https://github.com/neondatabase/neon/commit/d8c69e7c5dab097404d3800284044030bb4ed5a5 says "orphan layers are not errors". the idea is that we allow the orphan layers which are unlinked layers.

Reply:
Hm, I haven't followed scrubber too closely.
I.e., I don't know whether scrubber distinguishes between errors and "expected garbage".
Did @jcsp review this piece?

Reply:
It's okay to drop this -- the original code (not written by me) was basically asserting a totally quiescent state, where all layers were referenced by index and vice versa.  In real life, a layer can be un-referenced by the index during deletion, compaction, or initial creation.

Ideally we would warn on this case, or have a global flag for the  scrubber for how sensitive it should be (so that during certain tests we would run an "everything should be  perfect" mode).

Reply:
Yeah, that's how I remembered the plan for the scrubber: under normal operation, we shouldn't be leaking layers.
So, I guess we should have infra in the test framework to mark a test as "leaked layers are ok/expected".

@koivunej what tests are actually leaking layers with this PR?

Reply:
Probably the ones which are using immediate stop or restart... Checking.

Reply:
Weird, no errors this time. I did not remove it for fun, it was causing failed tests, but it is possible these were additional errors to tests which are now benefitting from the timeline_gc waiting.

Reply:
Left this out.